Kernel/VM/MemoryManager.cpp

```diff
@@ -60,13 +61,19 @@ void MemoryManager::initialize_paging()
     map_protected(VirtualAddress(0), PAGE_SIZE);
 
 #ifdef MM_DEBUG
-    dbgprintf("MM: Identity map bottom 8MB\n");
+    dbgprintf("MM: Identity map bottom 1MiB\n", kernel_virtual_base);
 #endif
-    // The bottom 8 MB (except for the null page) are identity mapped & supervisor only.
-    // Every process shares these mappings.
-    create_identity_mapping(kernel_page_directory(), VirtualAddress(PAGE_SIZE), (8 * MB) - PAGE_SIZE);
+    create_identity_mapping(kernel_page_directory(), VirtualAddress(PAGE_SIZE), (1 * MB));
 
-    // FIXME: We should move everything kernel-related above the 0xc0000000 virtual mark.
+#ifdef MM_DEBUG
+    dbgprintf("MM: Map kernel at %p\n", kernel_virtual_base);
+#endif
+    // Map the kernel to its' higher address range
+    create_mapping(kernel_page_directory(), PhysicalAddress(PAGE_SIZE), VirtualAddress(kernel_virtual_base + PAGE_SIZE), false, true, true, (8 * MB) - PAGE_SIZE);
```

Comment:
So here we're just re-doing the work that the initial bootup code already did?

Reply:
Yeah we technically are. My reasoning was that we need to identity map more of the kernel at boot otherwise it'll crash, because the next instruction is a physical address that woulnd't be mapped with the mapping that `MemoryManager::initialize()` creates. Though, on thought, I could just remove that in `boot.S` so I'll move it all there.